packages/flutter/lib/src/rendering/box.dart

```diff
@@ -2466,12 +2477,12 @@ mixin RenderBoxContainerDefaultsMixin<ChildType extends RenderBox, ParentDataTyp
   ///
   /// Useful when the children are displayed vertically in the same order they
   /// appear in the child list.
-  double defaultComputeDistanceToFirstActualBaseline(TextBaseline baseline) {
-    assert(!debugNeedsLayout);
-    ChildType child = firstChild;
+  double? defaultComputeDistanceToFirstActualBaseline(TextBaseline baseline) {
+    assert(!debugNeedsLayout!);
+    ChildType? child = firstChild;
     while (child != null) {
       final ParentDataType childParentData = child.parentData as ParentDataType;
```

Comment:
how did _parentData_ end up being non-nullable? o_O

Reply:
I perhaps made it non-nullable because it need to be non-nullable in the follow code to work.

Reply:
Changed to nullable and added `!`

Reply:
again i'm just surprised the tests passed and the code compiled... :-/

Reply:
cc @leaf @lrhn this is the kind of thing that made me file https://github.com/dart-lang/language/issues/1170